tests/ui.rs

```diff
@@ -0,0 +1,10 @@
+use trybuild::TestCases;
+
+#[test]
+fn ui() {
+    let t = TestCases::new();
+
+    t.pass("tests/ui/00-initial-concept.rs");
+    t.pass("tests/ui/01-tupled-struct.rs");
+    t.pass("tests/ui/01-named-struct.rs");
```

Comment:
Maybe consider using [the wildcard](https://docs.rs/trybuild/latest/trybuild/#compile-fail-tests)

Reply:
The problem with the wildcard is that it requires us to have two naming conventions depending on if the test should compile or not, so that both wildcards don't overlap.

Reply:
A declarative way to do this would be to use a prefix in the filenames:
```rust
t.pass("tests/ui/pass_*.rs");
t.compile_fail("tests/ui/fail_*.rs");
```
Even though this creates a "division" in the naming of the tests, I find it better than to (manually, exhaustively) hardcode the filenames in the harness launcher.

Reply:
That's correct and super cool, but that's enough bikeshedding for a PR which will include at maximum 4 test cases.